intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -65,28 +66,26 @@ class MirrordConfigDropDown : ComboBoxAction() {
         e.project?.let { project ->
             // this check ensures that we don't query the index when it is being built
             // querying the index during the startup/Dumb Mode can give us 0 or stale data
-            if (!::configFiles.isInitialized || updateConfigs.getAndSet(false)) {
-                queryIndexInSmartMode(project) {
-                    updateConfigFiles(project)
-                }
+            if (!::configFiles.isInitialized) {
+                e.presentation.isVisible = false
+                blockAndQueryIndex(project)
                 return
             }
 
-            val files = configFiles
-            if (files.size < 2) {
-                chosenFile = files.firstOrNull()
-                e.presentation.isVisible = false
-                return
+            if (updateConfigs.getAndSet(false)) {
+                blockAndQueryIndex(project)
             }
 
-            chosenFile = chosenFile ?: files.first()
+            if (configFiles.size > 1) {
+                if (chosenFile !in configFiles) {
+                    chosenFile = configFiles.first()
+                }
 
-            if (chosenFile !in files) {
-                chosenFile = files.first()
+                e.presentation.text = chosenFile?.let { getReadablePath(it, project) }
+            } else {
+                chosenFile = configFiles.firstOrNull()
             }
-
-            e.presentation.text = getReadablePath(chosenFile!!, project)
-            e.presentation.isVisible = true
+            e.presentation.isVisible = configFiles.size > 1
```

Comment:
No need to do all this if `blockAndQueryIndex` was not called, right? Why not have 1 if for both the not-initialized-yet case and the `updateConfig` case, and in it call `blockAndQueryIndex` and then do all of this?

Reply:
initialization needs to set `e.presentation.isVisible = false`

Reply:
> initialization needs to set `e.presentation.isVisible = false`

why?

Reply:
why show the config dropdown when there are no configs?

Reply:
How do you know there are no configs? The hash map is not initialized yet, but then you call the function that updates them, so now they may or may not be values in the HashMap, and you can just have the same logic as after an update - if there are (more then 1) configs - show the menu, otherwise - dont.

Reply:
you get the configFiles not initialized exception, would need further synchronization

Reply:
Where do you get that exception?
After calling the function that updates them they are initialized, no?

Reply:
if you dont return and try to access the configFiles

Reply:
remember, the update is a task we don't know if the hashmap has been initialized